arch/src/x86_64/acpi.rs

```diff
@@ -151,6 +151,14 @@ pub fn create_dsdt_table(
         0x00, 0x00, 0x00, 0x08, 0x00, 0x00, 0x00, 0x79, 0x00,
     ];
 
+    // Patch Range Minimum/Range Maximum/Length for the the 64-bit device area
+    pci_dsdt_data[170..174].copy_from_slice(&layout::MEM_32BIT_DEVICES_START.0.to_le_bytes()[0..4]);
+    pci_dsdt_data[174..178].copy_from_slice(
+        &(layout::MEM_32BIT_DEVICES_START.0 + layout::MEM_32BIT_DEVICES_SIZE - 1).to_le_bytes()
```

Comment:
Is `start + size - 1` the right thing to do here? I thought `Range Maximum` was about the last address that you can use to access some memory, and I thought ACPI was reading 4 bytes at a time?

Reply:
It is the last addressable byte so if we have an offset (the start address) and then add a size, we get an offset which would be the next usable byte. But that's not what we want here, so to get the last byte in the range we have to subtract zero.

Honestly, it's a bit annoying that this requires, first, last and length since you only actually need  2 of the three.

Reply:
Yeah...